tests/unit/test_dataframe.py

```diff
@@ -180,3 +180,22 @@ def test_create_or_replace_temp_view_bad_input():
         "The input of create_or_replace_temp_view() can only a str or list of strs."
         in str(exc_info)
     )
+
+
+def test_same_joins_should_generate_same_queries():
```

Comment:
You can just run it as an integration test? Also, could you check whether q1.join(q2) works?